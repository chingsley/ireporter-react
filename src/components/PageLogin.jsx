import React from 'react';
import FormLogin from './FormAuth';

const LoginPage = props => <FormLogin formType="login" {...props} />;

export default LoginPage;
