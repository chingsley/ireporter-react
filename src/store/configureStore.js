import {
  applyMiddleware, createStore, combineReducers, compose,
} from 'redux';
import thunk from 'redux-thunk';
import authReducer from '../reducers/authReducer';

export default () => {
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line
  const store = createStore(
    combineReducers({
      auth: authReducer,
    }),
    composeEnhancers(applyMiddleware(thunk)),
  );
  return store;
};
