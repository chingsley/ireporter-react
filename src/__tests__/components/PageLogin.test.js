import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import PageLogin from '../../components/PageLogin';

const wrap = () => {
  const props = {
    history: {
      push: jest.fn(),
      location: {
        pathname: '',
      },
    },

    authorise: jest.fn(),
    isLoggedIn: false,
    fetching: false,
    formType: 'signup',
  };

  const mockStore = configureStore([thunk]);

  const store = mockStore({
    auth: {
      isLoggedIn: true,
      user: {},
    },
    fetchStatus: {
      fetching: false,
      errMessage: '',
    },
    report: {
      report: {},
      userReports: [
        {
          id: 1, comment: 'some comments here', location: '1.2, 2.1', status: 'draft', createdOn: Date.now(), Images: ['a-given-image.png'],
        },
        {
          id: 2, comment: 'some comments here', location: '1.2, 2.1', status: 'draft', createdOn: Date.now(), Images: ['another-img.npn'],
        },
      ],
    },
  });

  const shallowWrapper = shallow(<PageLogin {...props} />);

  const mountWrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <PageLogin {...props} />
      </MemoryRouter>
    </Provider>,
  );
  return { shallowWrapper, mountWrapper };
};

describe('Test the report form', () => {
  const { mountWrapper: wrapper } = wrap();
  it('should render with the correct elements', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
