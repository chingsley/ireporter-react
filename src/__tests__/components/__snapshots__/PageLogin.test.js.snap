// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the report form should render with the correct elements 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "0vaiga",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "0vaiga",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <LoginPage
        authorise={[MockFunction]}
        fetching={false}
        formType="signup"
        history={
          Object {
            "location": Object {
              "pathname": "",
            },
            "push": [MockFunction],
          }
        }
        isLoggedIn={false}
      >
        <Connect(FormAuth)
          authorise={[MockFunction]}
          fetching={false}
          formType="signup"
          history={
            Object {
              "location": Object {
                "pathname": "",
              },
              "push": [MockFunction],
            }
          }
          isLoggedIn={false}
        >
          <FormAuth
            authorise={[Function]}
            fetching={false}
            formType="signup"
            history={
              Object {
                "location": Object {
                  "pathname": "",
                },
                "push": [MockFunction],
              }
            }
            isLoggedIn={true}
          >
            <div
              className="container"
            >
              <form
                className="container__form"
                onSubmit={[Function]}
              >
                <input
                  className="container__form-input"
                  id="email"
                  onChange={[Function]}
                  placeholder="email"
                  required={true}
                  type="email"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="password"
                  onChange={[Function]}
                  placeholder="password"
                  required={true}
                  type="password"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="firstname"
                  onChange={[Function]}
                  placeholder="first name"
                  required={true}
                  type="text"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="lastname"
                  onChange={[Function]}
                  placeholder="last name"
                  required={true}
                  type="text"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="othernames"
                  onChange={[Function]}
                  placeholder="other names"
                  type="text"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="username"
                  onChange={[Function]}
                  placeholder="username"
                  type="text"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="phonenumber"
                  onChange={[Function]}
                  placeholder="phone number"
                  required={true}
                  type="text"
                  value=""
                />
                <input
                  className="container__form-input"
                  id="adminSecret"
                  onChange={[Function]}
                  placeholder="admin secret key"
                  type="text"
                  value=""
                />
                <button
                  className="container__form-btn"
                  id="btn-reg"
                  type="submit"
                >
                  REGISTER
                </button>
              </form>
            </div>
          </FormAuth>
        </Connect(FormAuth)>
      </LoginPage>
    </Router>
  </MemoryRouter>
</Provider>
`;
